### ISSUE TITLE
Bad Request Error when has so many items and regions, with too much data.

### ISSUE THREAD

**Issue by igorsala7** [Dani3lSun/apex-plugin-apexscreencapture#3]
An error occured when I clicked on the button to screenshot to Item. I saw that it happens only if the option is "save to Item" and the page is full of items, regions, etc... with too much data. Do you know what could be? Thank you again.

1 step:

2 step:

3 step:

**Comment by Dani3lSun** [comment]
Hi @igorsala7 
can you please include screenshots of APEX Page Designer? From the plugin settings and the Item settings would be nice!

Thanks!

**Comment by Dani3lSun** [comment]
I also tested it with a larger page without problems, see the screencast:
![screencast](https://cloud.githubusercontent.com/assets/698942/12092987/acc03b76-b300-11e5-8da1-3358211c86e6.gif)

My plugin settings look like this:
<img width="289" alt="plg_settings" src="https://cloud.githubusercontent.com/assets/698942/12092910/497489fa-b300-11e5-9d84-09ead90339ba.png">

And my item settings like this: (important: Value protected to No)
<img width="292" alt="item_settings" src="https://cloud.githubusercontent.com/assets/698942/12092929/6aaa39a8-b300-11e5-9750-f0e9a674995e.png">

**Comment by igorsala7** [comment]
Ok, there is about 102 regions, 100 graphs regions on a pl/sql view returning about 20000 results each. I read about something appointing to some cache problem.

![error_part_4](https://cloud.githubusercontent.com/assets/16291851/12093456/32b12252-b2ea-11e5-9410-1a2ab4e3bb3b.png)

**Comment by igorsala7** [comment]
![error_part_5](https://cloud.githubusercontent.com/assets/16291851/12093846/6dd7dd1a-b2ec-11e5-9b96-7fafc642363d.png)

**Comment by Dani3lSun** [comment]
Ok thats a huge number of regions...But tested it with 45-50 regions and the result was fine. The resulting png had 4MB.
Can you test it if you select only one region instead of the whole inner body? One graph or the title bar...Does it work then?

**Comment by igorsala7** [comment]
Yes, it works fine with little regions in specific, now, if I test on a region with many sub-regions, crash. I think may be some problem to cache all the area, because of the many results of query, my page is so loaded.

**Comment by Dani3lSun** [comment]
btw if you want to test if writing to an item works, you can create a DA on Change of the Item --> execute javascript --> code:

```
var pngtext = $v('P15_IMAGE_HOLDER');
alert(pngtext);
window.open(pngtext,'_blank');
```

**Comment by Dani3lSun** [comment]
OK strange, maybe the huge number of graphs within subregions, confusing...^^
Does it work without the UI selector?
DOM UI Selector --> NO
JQuery selector --> body
And writing to item

**Comment by igorsala7** [comment]
hahahaha little confusing

I put the javascript alert that you said, and changed to DOM UI Selector --> NO, JQuery selector --> body. Same error, doesn't show the JS alert, but the Bad Request Error shine hahaha.

**Comment by igorsala7** [comment]
I changed to open in a new tab and works fine, but, I need to save this image in a blob. A specific region works fine, like the graph "Faixa de Idade", which is a sub-region of "Indicadores: Dependentes".

**Comment by Dani3lSun** [comment]
OK at least writing the whole body without UI selector gives the same error, so the UI selector works fine and the error is somewhere else...
1) I think maybe

```
apex.item( "P1_ITEM" ).setValue
```

has a maximum size or browsers can´t insert endless data in a hidden input field...
2) Or the canvas object itself gets too big:
http://stackoverflow.com/questions/6081483/maximum-size-of-a-canvas-element
But then opening in a new tab shouldn´t work either...but that works, so problem is No 1 I think...
I will come back to you if I have more Infos!

**Comment by igorsala7** [comment]
Ok, thank you! I'll search for some information too.

**Comment by Dani3lSun** [comment]
Ok talked to Dan McGhan,

the most realistic error is the POST max size of the web server. On which one runs your instance?
You have to increase that size so Javascript can save it to the item. Tomcat for example has a default of 2MB, that is too less...Maybe 8-10MB suits for your big page...

**Comment by Dani3lSun** [comment]
If you save the screenshot to a tab, when downloading the file, you see the actual size of the file...

**Comment by igorsala7** [comment]
We use Oracle HTTP Server (Apache) and mod_plsql. That bigger image I uploaded here has about 530Kb. I'll see with our DBA something. Thank you!

**Comment by Dani3lSun** [comment]
for OHS this parameter should be inspected:

```
LimitRequestBody
LimitXMLRequestBody
LimitRequestLine
```

edit: but it should have something todo with your server config, on my server I can save a nearly 4MB png into a item...
screencapture functions work well, otherwise opening in a new tab shouldn´t work either...

**Comment by Dani3lSun** [comment]
Hi @igorsala7 

the error was indeed on the plugin side, if writing to an item not more than 32k is possible.
Tested it with some screenshots, all images in DB was cut by 32k and only the above content was visible...
But good news for you I released version 1.5, which no longer saves the image to an item, instead it saves the images using AJAX and a custom PL/SQL procedure to DB. A default procedure shows how it works with BLOBs!
https://github.com/Dani3lSun/apex-plugin-apexscreencapture/releases/tag/v1.5
- Daniel
